rasa/shared/core/events.py

```diff
@@ -103,6 +102,69 @@ def first_key(d: Dict[Text, Any], default_key: Any) -> Any:
         return None
 
 
+def split_events(
+    events: Iterable["Event"],
+    event_type_to_split_on: Type["Event"],
+    additional_splitting_conditions: Optional[Dict[Text, Any]] = None,
+    include_splitting_event: bool = True,
+) -> List[List["Event"]]:
+    """Splits events according to an event type and condition.
+
+    Examples:
+        Splitting events according to the event type `ActionExecuted` and the
+        `action_name` 'action_session_start' would look as follows:
+
+        >> _events = split_events(
+                        events,
+                        ActionExecuted,
+                        {"action_name": "action_session_start"},
+                        True
+                     )
+
+    Args:
+        events: Events to split.
+        event_type_to_split_on: The event type to split on.
+        additional_splitting_conditions: Additional event attributes to split on.
+        include_splitting_event: Whether the events of the type on which the split
+            is based should be included in the returned events.
+
+    Returns:
+        The split events.
+    """
+    sub_events = []
+    current = []
+
+    def event_fulfills_splitting_condition(evt: "Event") -> bool:
+        # event does not have the correct type
+        if not isinstance(evt, event_type_to_split_on):
+            return False
+
+        # the type is correct and there are no further conditions
+        if not additional_splitting_conditions:
+            return True
+
+        # there are further conditions - check those
+        return all(
+            getattr(evt, k) == v for k, v in additional_splitting_conditions.items()
```

Comment:
That would avoid an `AttributeError` in case the event for some reason doesn't have this attribute yet.
```suggestion
            getattr(evt, k, None) == v for k, v in additional_splitting_conditions.items()
```